### ISSUE TITLE
[Bug] "External connection exception occurred: not a gzip file" error when running dbt deps

### ISSUE THREAD

**Issue by Visya** [dbt-labs/dbt-core#9063]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

When running dbt deps behind proxy (with `HTTP_PROXY`, `HTTPS_PROXY` and `FTP_PROXY` variables set up), I get an error:
```
External connection exception occurred: not a gzip file
```

I tried installing with git, which works, but as soon as I try to install any bigger package that depends on `dbt_utils`, it fails with  "Found duplicate project "dbt_utils". This occurs when a dependency has the same project name as some other dependency."

### Expected Behavior

Packages are able to be installed behind the proxy with dbt deps.

### Steps To Reproduce

1. Set `HTTP_PROXY`, `HTTPS_PROXY` environment variables.
2. Add a package to packages.yml.
```
packages:
  - package: dbt-labs/dbt_utils
    version: 1.1.1
```
4. Run `dbt deps`.
5. Observe error "External connection exception occurred: not a gzip file".

### Relevant log output

```shell
11:17:28  Running with dbt=1.7.1
11:17:29  Updating lock file in file path: [...]code\test\src/package-lock.yml
11:17:30  Installing dbt-labs/dbt_utils
11:17:39  Encountered an error:
External connection exception occurred: not a gzip file
```

### Environment

```markdown
- OS: Windows 10 Enterprise
- Python: 3.11.5
- dbt: 1.7.1
```

### Which database adapter are you using with dbt?

postgres

### Additional Context

I have `python-certifi-win32` installed, which resolved issues similar to [this issue](https://github.com/dbt-labs/dbt-core/issues/7153).

**Comment by jtcohen6** [comment]
Hey @Visya! I see you're using v1.7.1 — is this working for you on earlier versions of `dbt-core`? We need to understand if this is a regression (due to some change in v1.7), versus that has never worked.

**Comment by Visya** [comment]
Hi @jtcohen6, I am setting up a new environment in a new system, so it never worked for me. But I have other colleagues in the same network, that are able to download packages behind the proxy.

**Comment by graciegoheen** [comment]
@Visya do you happen to know which version of dbt your colleagues are using?

**Comment by josezeta** [comment]
Currently we are running some workflows using DBT-core v 1.7.4 in which this error is happeing.

**Comment by jfo8001** [comment]
I saw this error for the first time today having upgraded our snowflake connector recently.   DBT core has remained the same for the last 9 months. 
dbt-core = "^1.5.2"
dbt-snowflake = "^1.5.2"
snowflake-connector-python = "^3.7.1"

**Comment by dbeatty10** [comment]
@josezeta and @jfo8001 did each of you see this error for the first time today?

If so, was it intermittent, or did it happen every time to you ran `dbt deps`? Are you using a proxy like the original poster, by any chance?

Also, could you share your environment details?

```
- OS:
- Python:
- dbt:
```

Examples:

**OS**: Ubuntu 20.04
**Python**: 3.9.12 (`python3 --version`)
**dbt-core**: 1.1.1 (`dbt --version`)

**Comment by josezeta** [comment]
@dbeatty10  It is intermittent, it has happend just one time today, 3 min after that fail the dbt was executed with no issues. It has been running each 15 min since them with no errors.

The error appears in 1 dbt deps execution with no code changes or any other environment change that why I think it maybe related with dbt dbt not being able to download the package's .gzip artifact due to some of transient issue in Github (That as far i know if the artifact reposotory)

**Comment by dbeatty10** [comment]
A transient issue with GitHub sounds like a very possible explanation.

Agreed that this doesn't look like an issue in dbt-core, and it being rare and intermittent both supporting that judgement.

I'm going to leave this as `awaiting_reponse` in case anyone adds more information that we should consider.

**Comment by jturner18** [comment]
Wanted to add I also am dealing with the "External connection exception occurred: not a gzip file"
and occasionally a rare "External connection exception occurred: Compressed file ended before the end-of-stream marker was reached" error.

OS: Microsoft Windows 10 Enterprise - 10.0.19045 Build 19045
Python: 3.11.9
DBT: 1.7.13
Snowflake: 1.7.3

**Comment by jturner18** [comment]
I think it is specifically a git issue with dbt-labs/dbt-utils (not all packages)

Even if I reference link: https://hub.getdbt.com/dbt-labs/dbt_utils/1.1.1/
and put code:

- package: dbt-labs/dbt_utils
    version: 1.1.1
    
in packages.yml -> get error:
Version error for package dbt-labs/dbt_utils: Could not find a satisfactory version from options: ['=1.1.1', '>=0.8.0', '<2.0.0', '>=0.8.0', '<0.9.0', '>=0.8.1', '<0.9.0']

if try code:
- package: dbt-labs/dbt_utils
    version: [">=0.8.0", "<0.10.0"]
    
in packages.yml -> get error:
External connection exception occurred: not a gzip file

If I comment out only the dbt_utils reference, all other packages install fine, but then packages-lock.yml still adds dbt-labs/dbt-utils to the end of the file in which it errors. 

If any advice available let me know.

**Comment by dbeatty10** [comment]
@jturner18 are you still experiencing this issue?

Both of the `packages.yml` files below worked for me when I ran this:
```shell
dbt deps --upgrade
```

```yaml
# one version of packages.yml 

packages:
  - package: dbt-labs/dbt_utils
    version: 1.1.1
```

```yaml
# another version of packages.yml 

packages:
  - package: dbt-labs/dbt_utils
    version: [">=0.8.0", "<0.10.0"]
```

**Comment by IL-Jerry** [comment]
I had the same issue with my dbt Cloud job running every 15 minutes. It happened twice and haven't happened since.
The same error message
```
------------------------------------------------------------
Invoke dbt Command
------------------------------------------------------------
dbt deps

23:15:14 Running with dbt=1.7.17
23:15:15 Installing dbt-labs/dbt_utils
dbt command failed23:15:35 Encountered an error:
External connection exception occurred: not a gzip file
```
**My config:**
- I don't use proxy
- My package.yml config:
```
packages:
    - package: dbt-labs/dbt_utils
      version: [">=1.1.0", "<1.2.0"]
    - package: fivetran/fivetran_log
      version: 1.4.3
    - package: dbt-labs/codegen
      version: 0.12.1
    - local: local-packages/dq_tools
    - package: Datavault-UK/automate_dv
      version: 0.10.0
    - local: local-packages/dbtvault_ext
    - package: entechlog/dbt_snow_mask
      version: 0.2.3
    - local: local-packages/finops
```
- This issue happened only with dbt-labs/dbt_utils so far.

Can someone take a look at this please? Thanks.

**Comment by dbeatty10** [comment]
@IL-Jerry Our best guess is that this is a caused by transient issue with GitHub.

**Comment by IL-Jerry** [comment]
Thanks @dbeatty10 I'm waiting an official explanation from someone from dbt Labs to explain to my clients.

**Comment by dbeatty10** [comment]
@IL-Jerry I'm one of the maintainers of dbt-core. This error message has been reported several times within this GitHub issue. Each time it has been intermittent, and we haven't been able to reproduce the issue ourselves.

https://github.com/dbt-labs/dbt-core/issues/4579 was opened by Scott Barber, who lead the dbt Cloud support team for several years. He wrote that this type of error is "typically ... indicative of a transient problem with github itself".

If you'd like further support, you can reach out to the dbt Cloud Support team via the dbt Cloud web interface (see below for screenshots). If so, please include a link to this issue when you create the support ticket.

### Classic navigation for dbt Cloud

Click question mark icon (?) > Create a support ticket

<img width="375" alt="image" src="https://github.com/user-attachments/assets/12618090-0b49-4cf7-8638-0d11a6df8102">

### Newest navigation interface for dbt Cloud:

Help & Guides > Create a support ticket

<img width="500" alt="image" src="https://github.com/user-attachments/assets/6b275809-1f44-4dbf-a76e-32d9815d1a16">

### More detail

[Here](https://github.com/dbt-labs/dbt-common/blob/35af6540b1f4a1c58cac895e6205a4092383e6e3/dbt_common/utils/connection.py#L11-L36) is the source code that raises that exception. It will re-attempt the download up to 5 times with 1 second wait in between each attempt. After that, it will raise an exception that starts with "External connection exception occurred: " followed by the original exception message.

**Comment by dbeatty10** [comment]
@IL-Jerry I connected with our dbt Cloud Support team.

They shared that [GitHub had an outage](https://www.githubstatus.com/incidents/kz4khcgdsfdv) August 14, 2024 between 23:02 UTC and 23:38 UTC. So if your clients were experiencing this issue in that time frame, then that would be consistent with this GitHub outage.

The GitHub outage affected dbt Cloud services, and we [reported this](https://status.getdbt.com/incidents/nplqjd72dhd5) on our [status page](https://status.getdbt.com/history) as well.

If you have any other questions or feedback, please reach out to dbt Cloud Support team, and they'd be glad to help.

**Comment by IL-Jerry** [comment]
Thanks @dbeatty10 for the detailed response.

**Comment by pranjalbhatt** [comment]
Hi, 

I am receiving the same error  `ConnectionError: External connection exception occurred: not a gzip file` while downloading packages via `packages.yml` file.

My file looks like- 

`
packages:
  - package: calogica/dbt_expectations
    version: 0.10.4
  - package: brooklyn-data/dbt_artifacts
    version: 2.6.4
  - package: elementary-data/elementary
    version: 0.16.1
`

Error only occurs while downloading `elementary` package, rest of the packages get installed.

**Comment by dbeatty10** [comment]
@pranjalbhatt are you still seeing this error? Or did it resolve?

**Comment by pranjalbhatt** [comment]
> @pranjalbhatt are you still seeing this error? Or did it resolve?

Yes. I am still seeing this error. Unable to download `dbt-utils` and `elementary`. Tried to change versions as well. Is there a fix ?

**Comment by dbeatty10** [comment]
@pranjalbhatt I am able to successfully install the `elementary` version `0.16.1` using `dbt deps` via both dbt-core 1.7 and 1.8.

Can you share the output of `dbt --version` so I can see which version of dbt-core you are using?

**Comment by pranjalbhatt** [comment]
> @pranjalbhatt I am able to successfully install the `elementary` version `0.16.1` using `dbt deps` via both dbt-core 1.7 and 1.8.
> 
> Can you share the output of `dbt --version` so I can see which version of dbt-core you are using?

Yes, still unable to download the elementary. I am using dbt-core. 

`Core:
  - installed: 1.8.1` 

As a temporary fix, I have used `git clone` to do install the package

**Comment by dbeatty10** [comment]
@pranjalbhatt I am not able to replicate the issue you are having, and I suspect it is distinct from the other reports in this issue. If you are continuing to experience this, could you open a new issue [here](https://github.com/dbt-labs/dbt-core/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=bug-report.yml&title=%5BBug%5D+%3Ctitle%3E)?

**Comment by ae89387** [comment]
We are having similar issue as well - 

![Image](https://github.com/user-attachments/assets/d41b29f1-22e4-41ce-b7b6-83fb66c90e66)

**Comment by graciegoheen** [comment]
Closing this, as it seems to be caused by Github outages.
